compiler/fir/analysis-tests/testData/extendedCheckers/VariableAssignmentChecker.kt

```diff
@@ -0,0 +1,135 @@
+// WITH_RUNTIME
+
+import kotlin.properties.Delegates
+
+fun withAnnotation(p: List<Any>) {
+    @Suppress("UNCHECKED_CAST")
+    <!CAN_BE_VAL!>var<!> v = p as <!REDUNDANT_EXPLICIT_TYPE!>List<String><!>
+    print(v)
+}
+
+fun withReadonlyDeligate() {
+    val s: String by lazy { "Hello!" }
+    s.hashCode()
+}
+
+fun destructuringDeclaration() {
+    var (<!CAN_BE_VAL!>v1<!>, <!CAN_BE_VAL!>v2<!>) = getPair()
+    print(v1)
+
+    var (<!CAN_BE_VAL!>v3<!>, v4) = getPair()
+    print(v3)
+    v4 = ""
+}
+
+fun getPair(): Pair<Int, String> = Pair(1, "1")
+
+fun stackOverflowBug() {
+    <!CAN_BE_VAL!>var<!> a: Int
+    a = 1
+    for (i in 1..10)
+        print(i)
+}
+
+fun listReceiver(p: List<String>) {}
+
+fun withInitializer() {
+    var v1 = 1
+    var v2 = 2
+    <!CAN_BE_VAL!>var<!> v3 = 3
+    v1 = 1
+    v2++
+    print(v3)
+}
+
+fun test() {
+    var a = 0
+    while (a>0) {
+        a++
+    }
+}
+
+fun foo() {
+    <!CAN_BE_VAL!>var<!> a: Int
+    val bool = true
+    <!UNINITIALIZED_VARIABLE_EXTENDED!>val<!> b: String
```

Comment:
Why EXTENDED?

Reply:
I use this name cuz just `UNINITIALIZED_VARIABLE` already in use in standart checker and it is an `error`